File: MCTabBarDemo/MCTabBar/BaseNavigationController.m
//
//  BaseNavigationController.m
//  MCTabBarDemo
//
//  Created by chh on 2017/12/18.
//  Copyright © 2017年 Mr.C. All rights reserved.
//

#import "BaseNavigationController.h"

@interface BaseNavigationController ()

@end

@implementation BaseNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    //可以设置一些样式
    /*
     //设置了NO之后View自动下沉navigationBar的高度
     self.navigationBar.translucent = NO;
     //改变左右Item的颜色
     self.navigationBar.tintColor = [UIColor whiteColor];
     
     //改变title的字体样式
     NSDictionary *textAttributes = @{NSForegroundColorAttributeName : [UIColor whiteColor], NSFontAttributeName: FONT_SYSTEM_BOLD(18)};
     [self.navigationBar setTitleTextAttributes:textAttributes];
     //改变navBar的背景颜色
     [self.navigationBar setBarTintColor:[UIColor blueColor]];
     */
}

//重写这个方法，在跳转后自动隐藏tabbar
- (void)pushViewController:(UIViewController *)viewController animated:(BOOL)animated
{
    if ([self.viewControllers count] > 0){
        viewController.hidesBottomBarWhenPushed = YES;
        //可以在这里定义返回按钮等
        /*
        UIBarButtonItem *backItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"nav_back"] style:UIBarButtonItemStylePlain target:self action:@selector(backAction)];
        viewController.navigationItem.leftBarButtonItem = backItem;
         */
    }
    //一定要写在最后，要不然无效
    [super pushViewController:viewController animated:animated];
    CGRect rect = self.tabBarController.tabBar.frame;
    rect.origin.y = [UIScreen mainScreen].bounds.size.height - rect.size.height;
    self.tabBarController.tabBar.frame = rect;
}


@end

File: MCTabBarControllerSwift/BaseNavigationController.swift
//
//  BaseNavigationController.swift
//  MCTabbarDemoSwift
//
//  Created by caohouhong on 2018/12/5.
//  Copyright © 2018年 caohouhong. All rights reserved.
//

import UIKit

class BaseNavigationController: UINavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let navBar = UINavigationBar.appearance()
        //设置false之后自动下沉navigationBar 高度
        self.navigationBar.isTranslucent = true
        //设置navigationBar的背景色
        navBar.barTintColor = UIColor.white
        //设置左右bar的颜色
//        navBar.tintColor = UIColor.red
//        // 设置标题的样式
//        let attriDic = [NSAttributedStringKey.font:UIFont.systemFont(ofSize: 18),NSAttributedStringKey.foregroundColor:UIColor.red]
//        navBar.titleTextAttributes = attriDic
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        
    }
    
    override func pushViewController(_ viewController: UIViewController, animated: Bool) {
        // 不是第一级vc
        if viewControllers.count > 0 {
            viewController.hidesBottomBarWhenPushed = true
            // 可以在这里设置返回按钮等,重写后侧滑返回失效需要自己单独处理
            //backBarButtonItem 是带有字和返回箭头的样式
            /*
            viewController.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "nav_back_13x21"), style: .plain, target: self, action: #selector(navBackAction))
            */
        }
        //一定要写在最后，要不然无效
        super.pushViewController(viewController, animated: true)
    }

    // 返回操作
    @objc func navBackAction(){
        popViewController(animated: true)
    }

}
